app/models/user.rb

```diff
@@ -25,7 +25,7 @@ def can_edit_site?(site_to_edit)
         own_organisation == site_to_edit.organisation ||
         site_to_edit.organisation.parent_organisations.include?(own_organisation) ||
         site_to_edit.extra_organisations.include?(own_organisation) &&
-        site_to_edit.global_type.blank?
+          site_to_edit.global_type.blank?
```

Comment:
Aha! The intent is `(a || b || c || d) && e` - `e` is only about the site whereas all the others are about the privileges of the user in relation to the site.

In practice it isn't possible to navigate to a page to add mappings for a globally archived/redirected site, but if you know the URL then it is possible to create mappings for them - I've just done it in integration for directgov_localadmin :-/ Those mappings won't be used for a site with a global status, but it shouldn't be possible. We should check that fixing the logic here fixes that bug.

I don't think this is a security risk because it still restricts users to editing mappings for sites they should be allowed to edit. Any mappings created via this bug would only be used by Bouncer if we removed the global status from the site, in which case we would have to sort out the mappings in advance anyway (I don't think we've ever done this).